explorer-ui/src/components/ProfileView/ProfileDisplay/ProfileDisplay.jsx

```diff
@@ -1,17 +1,136 @@
+          profiles[i].accomodation == userProfile.accomodation
+        ) {
+          profilesToAdd.add(profiles[i])
+        }
+      }
+      setMatches([...matches, ...profilesToAdd])
+      
+    };
+  
+      // Calculate the intersections between interests and attractions
+    function findIntersections(match){
+      if(match.preferenceInfo.interests && userProfile.preferenceInfo.interests){
+        let interstIntersection = userProfile.preferenceInfo.interests.filter(x => match.preferenceInfo.interests.includes(x));
+        let attractionIntersection = userProfile.preferenceInfo.attractions.filter(y =>  match.preferenceInfo.attractions.includes(y));
+        return {"interestIntersection": interstIntersection, "attractionIntersection": attractionIntersection, "match": match};
+      }
+      return {"interestIntersection": {}, "attractionIntersection": {}, "match": match};
+    }
+    
+    // Rank the matches based off number of intersections in descending order 
+    const rankMatches = (matches) =>{
+      if(matches){
+      // Create a state to keep track of the intersection data and iterate through each match to get that data
+      const matchesToRank = []
+      matches.forEach((match) => matchesToRank.push(findIntersections(match)))
+      matchesToRank.sort((a, b) => b.attractionIntersection.length - a.attractionIntersection.length)
+      setRankedMatches(matchesToRank)
+      }
+    }
+
+    // Get latitude & longitude position based off the address of each match
+    const getCoordinates = (rankedMatches) => {
+      if (rankedMatches) {
+        for (let i = 0; i < rankedMatches.length; i++) {
+          if(rankedMatches[i].match.address != null || rankedMatches[i].match.address !="" ){
+          
+          Geocode.fromAddress(rankedMatches[i].match.address).then(
+            (response) => {
+              const { lat, lng } = response.results[0].geometry.location;
+              setMatchesPlusCoordinates(matchesPlusCoordinates => [...matchesPlusCoordinates, {id: i, position: {lat, lng}, user: matches[i]}])
+              
+            },
+            (error) => {
+              console.error(error);
+            }
+          );
+        }
+        }
+        setCoordsLoaded(true);
+      }
+    };
+
+    // Get the latitude and longitude position of the logged in user to use as the center position in the map
+    function getCurrLocation(userProfile){
+      Geocode.fromAddress(userProfile.address).then(
+        (response) => {
+          const { lat, lng } = response.results[0].geometry.location;
+          setUserCoordinates({"string": userProfile.address, "position": {lat, lng}})
+        },
+        (error) => {
+          console.error(error);
+        }
+      );
+     
+    }
+
+  // Display the logged in users profile information, then a list of their matches, 
+  // and then a map of their matches locations
+  return (
+    <div className="profile-name">
+      {
+        userProfile !={}
```

Comment:
stricter check

Reply:
fixed- checking for empty object and null